common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -945,6 +959,92 @@ def get_course(self, course_key, depth=0, **kwargs):
         except ItemNotFoundError:
             return None
 
+    def do_index(self, location, delete=False):
+        """
+        Main routine to index (for purposes of searching) from given location and other stuff on down
+        """
+        error = []
+        # TODO - inline for now, need to move this out to a celery task
+        searcher = SearchEngine.get_search_engine(INDEX_NAME)
+        if not searcher:
+            return
+
+        location_info = {
+            "course": unicode(location.course_key),
+        }
+
+        def _fetch_item(item_location):
+            """ Fetch the item from the modulestore location, log if not found, but continue """
+            try:
+                item = self.get_item(item_location, revision=ModuleStoreEnum.RevisionOption.published_only)
+            except ItemNotFoundError:
+                log.warning('Cannot find: %s', item_location)
+                return None
+
+            return item
+
+        def index_item_location(item_location, current_start_date):
+            """ add this item to the search index """
+            item = _fetch_item(item_location)
+            if item:
+                if item.category in ['course', 'chapter', 'sequential', 'vertical', 'html', 'video']:
+                    if item.start and (not current_start_date or item.start > current_start_date):
+                        current_start_date = item.start
+
+                    if item.has_children:
+                        for child_loc in item.children:
+                            index_item_location(child_loc, current_start_date)
+
+                    item_index = {}
+                    try:
+                        item_index.update(location_info)
+                        item_index.update(item.index_dictionary())
+                        item_index.update({
+                            'id': unicode(item.scope_ids.usage_id),
+                        })
```

Comment:
It's silly and needlessly verbose to use `.update()` with just one key-value pair. Just do
```python
item_index['id'] = unicode(item.scope_ids.usage_id)
```